IPython/core/oinspect.py

```diff
@@ -766,23 +765,28 @@ def info(self, obj, oname='', formatter=None, info=None, detail_level=0):
         # Constructor docstring for classes
         if inspect.isclass(obj):
             out['isclass'] = True
-            # reconstruct the function definition and print it:
+
+            # get the function signature:
+            try:
+                init_def = self._getdef(obj, oname)
+            except AttributeError:
+                init_def = None
+            else:
+                out['init_definition'] = self.format(init_def)
+
+            # get the __init__ docstring
             try:
-                obj_init =  obj.__init__
+                obj_init = obj.__init__
             except AttributeError:
                 init_def = init_ds = None
```

Comment:
Should we remove the None-ing out of `init_def` here, since, this block is no longer setting that name?